src/editor.c

```diff
@@ -601,6 +601,7 @@ static void show_autocomplete(ScintillaObject *sci, gsize rootlen, GString *word
 	}
 	/* store whether a calltip is showing, so we can reshow it after autocompletion */
 	calltip.set = (gboolean) SSM(sci, SCI_CALLTIPACTIVE, 0, 0);
+	SSM(sci, SCI_AUTOCSETORDER, SC_ORDER_CUSTOM, 0);
```

Comment:
I read the docs but I couldn't grasp the difference between `SC_ORDER_PRESORTED` and `SC_ORDER_CUSTOM`. In either case scintilla does not sort by itself. `SC_ORDER_PRESORTED` seems to assume the list is presorted alphabetically but what's the deal with that?

Reply:
I didn't explore the logic of how we show the popup and if we need it (but @elextr seemed to experience some problem without it). The way I understand it is that when we call `SCI_AUTOCSHOW`, Scintilla takes care of filtering the list based on what the user types unless we update it ourselves. When the list is sorted, Scintilla can bisect where the start/end of the range to be displayed is and show the values in between (this really is my assumption, I haven't checked Scintilla sources). When not sorted, it first creates an index that corresponds to alphabetic ordering and does the bisection over this index. This makes the complexity `2*log(N)` (plus the initial creation of the index) instead of `N` if it had to go through all the values every time.

Reply:
thanks

Reply:
A quick scan of the code suggests Scintilla makes the index in all cases, but only sorts it in the `SC_ORDER_CUSTOM` case, so the sort is the extra cost.  Then as @techee said it binary searches the index for the selection each time you type, so when we passed `SC_ORDER_PRESORTED` by leaving the default it was binary searching an unsorted index and that gives undefined results, which is probably why @techee didn't see it and I did.